ebml_matroska.xml

```diff
@@ -332,7 +332,21 @@
   <element name="BlockAddIDType" path="\Segment\Tracks\TrackEntry\BlockAdditionMapping\BlockAddIDType" id="0x41E7" type="uinteger" minOccurs="1" maxOccurs="1" minver="4" default="0">
     <documentation lang="en" purpose="definition">Stores the registered identifer of the Block Additional Mapping to define how the BlockAdditional data should be handled.</documentation>
     <extension webm="0"/>
-  </element>
+    <restriction>
+      <enum value="0" label="BlockAddIDValue defines the content type">
+        <documentation lang="en" purpose="definition">This value indicates that BlockAddIDValue defines the content type, and a list of content type is in BlockAddID documentation.</documentation>
+      </enum>
+      <enum value="1" label="Codec specific content">
+        <documentation lang="en" purpose="definition">This value indicates that BlockAddIDExtraData defines the content type, and this definition depends on the Codec ID used. A transcoder SHOULD discard this content if the addition is not supported</documentation>
+      </enum>
+      <enum value="2" label="General purpose content">
+        <documentation lang="en" purpose="definition">This value indicates that BlockAddIDExtraData defines the content type, and this definition does not depend on the Codec ID used. A transcoder SHOULD NOT discard this content if the addition is not supported</documentation>
+      </enum>
+      <enum value="3" label="ISOBMFF-like extension">
+        <documentation lang="en" purpose="definition">This value indicates that BlockAddIDExtraData consists of a 32-bit type field and an arbitrary large content field. The values used for both fields are the same as used in header atoms in ISOBMFF files and its extensions such as the "mvcC" box in ISO 14496-15:2014. As it is not expected that ISOBMFF-like extension have frame related content, BlockAddIDValue SHOULD NOT be present.</documentation>
```

Comment:
> BlockAddIDValue SHOULD NOT be present

I'd change that to MUST NOT (yes, invalid file in that case). The question we've often heard from reviewers about "SHOULD (NOT)" is "when would it be OK to violate that rule? What should happen then?" and I see no situation in which we have anything useful to do with BlockAddIDValue in this case.